agent/grpc/client.go

```diff
@@ -94,19 +94,20 @@ func newDialer(servers ServerLocator, wrapper TLSWrapper) func(context.Context,
 				return nil, fmt.Errorf("TLS enabled but got nil TLS wrapper")
 			}
 
-			// Switch the connection into TLS mode
-			if _, err := conn.Write([]byte{byte(pool.RPCTLS)}); err != nil {
-				conn.Close()
-				return nil, err
-			}
-
-			// Wrap the connection in a TLS client
+			// Wrap the connection in a TLS client, return same conn if TLS disabled
```

Comment:
Is TLS is disabled shouldn't this entire block be skipped because of the `server.UseTLS` check above?

Reply:
Agree, I made a few weird tests, so, not sure about what is really happening.
The weird stuff is:
in dev mode with:
```
consul agent -dev -hcl 'rpc{enable_streaming=true}' -hcl use_streaming_backend=true -hcl 'ca_file="/etc/consul/ca.pem", cert_file="/etc/consul/cert.pem", key_file="/etc/consul/key.pem"'
```
=> for some reason it fixes the issue. But not sure yet about the why... trying to understand better.
I suspect that when running the streaming config on the server, the data is just proxified internally but is not encrypted... this part is a bit new to me, still looking

Reply:
@dnephin Ok, found it, this is linked to `verify_outgoing: true|false`

* With verify_outgoing = true => works
* with verify_outgoing = false => fails

To test, use master + launch client with:

```
consul agent -dev -hcl 'rpc{enable_streaming=true}' -hcl use_streaming_backend=true -hcl 'telemetry{prometheus_retention_time = "192h", disable_compat_1.9=true}' -hcl 'ca_file="/etc/consul/ca.pem", cert_file="/etc/consul/cert.pem", key_file="/etc/consul/key.pem", verify_outgoing=false'
```
a call to 
`curl -i "http://localhost:8500/v1/health/service/consul?wait=10m&index=12&pretty=true&stale"`

fails with `verify_outgoing=false`, succeeds with `verify_outgoing=true`

Reply:
@dnephin Found the issue, probably fixed properly in the next patch, but I had to do a bit more refactoring (essentially the tests)
Now works with both TLS and no TLS enabled